src/cube/transformation.rs

```diff
@@ -0,0 +1,107 @@
+use super::cube::{Coordinate, CubeState};
+use std::collections::HashMap;
+
+#[derive(PartialEq, Debug)]
+pub struct Transformation {
+    moves: HashMap<Coordinate, Coordinate>,
+}
+
+impl Transformation {
+    pub fn all_transformations() -> Vec<Transformation> {
+        Vec::new()
+    }
+
+    pub(super) fn apply(&self, state: &CubeState) -> CubeState {
+        [
+            self.apply_fo_face(state, 0),
+            self.apply_fo_face(state, 1),
+            self.apply_fo_face(state, 2),
+            self.apply_fo_face(state, 3),
+            self.apply_fo_face(state, 4),
+            self.apply_fo_face(state, 5),
+        ]
+    }
+
+    fn apply_fo_face(&self, state: &CubeState, face: usize) -> [[u8; 3]; 3] {
+        let mut new_face = [[0u8; 3]; 3];
+        for i in 0..3 {
+            for j in 0..3 {
+                let new_coordinate = (face, i, j);
+                let &(old_face, old_i, old_j) =
+                    self.moves.get(&new_coordinate).unwrap_or(&new_coordinate);
```

Comment:
By using `unwrap_or` the `&new_coordinate` acts as default. Is that what we really want to do in the case we do not find the `new_coordinate` in the `moves` map?

What is the intention behind getting the old coordinates from the `moves` map? Shouldn't we get it from latest cube state?